sjsonnet/src-jvm-native/sjsonnet/SjsonnetMain.scala

```diff
@@ -7,8 +7,24 @@ import java.nio.file.NoSuchFileException
 import scala.util.Try
 import scala.util.control.NonFatal
 
+// JsonnetWorker (in universe) extends this trait in the universe so that it can pass the cache based on Caffeine to main0 here
+trait ParseCacheInterface {
+  def getOrElseUpdate(key: (Path, String), defaultValue:  => Either[Error, (Expr, FileScope)]): Either[Error, (Expr, FileScope)]
+}
+
+// As JsonnetWorker (in universe) passes an object to main0 which extends the trait: ParseCacheInterface, class based on HashMap
+// which extends this trait is needed to use the default HashMap
+case class HashMapDefault () extends ParseCacheInterface {
```

Comment:
The comment is a bit too detailed. What you could say is something like `A default implementation based on a mutable HashMap. This implementation is not thread-safe.`

I'd make this a simple `class` (not a `case class`, because it's not "data" and we don't pattern match on it). The name could be `DefaultParseCache` or something like that. The name should reflect what the class does (it's a parse cache), not only how it's implemented.

Reply:
Yes, let's make this `ParseCache` and `DefaultParseCache`. Adding `Interface` to a name is an old Java practice that never caught on in Scala.

Reply:
Thank you for the review iulian and Stefan, I have updated them.